### PR TITLE
[IOPAY-240] outcome from authorizationCode

### PR DESCRIPTION
**Pull request by infantesimone**
#### List of Changes
- add `directAcquirer` in `WalletSession` to decide final `outcome`;
- add `TransactionResult` with:
   -  `OutcomeEnum`;
   - `NexiResultCodeEnum`;
   - `VposResultCodeEnum`;
- new mixpanel event `PAYMENT_OUTCOME_CODE`  with _outcome_ as custom property;

#### Motivation and Context

The goal of this PR is to determine the `outcome` starting from `authorizationCode (NexiResultCodeEnum or VposResultCodeEnum)` and `directAcquirer` (true with `NexiPos` and false with `Vpos`), to be able to send it to mixpanel and determine the final message;

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/response.pug [hunk 1]
```diff
@@ -22,34 +22,39 @@ html(lang='it')
                                 h3.text-center Operazione in corso
                                 .text-muted.text-center.my-4 Ti preghiamo di attendere, senza chiudere la pagina.
                         .windowcont__response
-                            .windowcont__response__detail(data-response="3") <!-- TX_ACCEPTED -->
+                            .windowcont__response__detail(data-response="0") <!-- SUCCESS -->
```
Comment (pasqualespica) [inline src/response.pug:25]: am I wrong or is some [status](https://pagopa.atlassian.net/wiki/spaces/I/pages/167216710/Mixpanel+Events#Verifica-payment-transaction---io-pay%2Fresponse.html) missing vs `PAYMENT_OUTCOME_CODE` ?

File: src/response.ts [hunk 2]
```diff
@@ -34,24 +34,26 @@ import {
...
 
 const config = getConfigOrThrow();
 
-const handleFinalStatusResult = (idStatus: GENERIC_STATUS, authorizationCode?: string) => {
-  const eventResult: string = getResultEventByAuthorizationCode(authorizationCode || '');
-  mixpanel.track(eventResult, {
-    EVENT_ID: eventResult,
+const handleFinalStatusResult = (idStatus: GENERIC_STATUS, authorizationCode?: string, isDirectAcquirer?: boolean) => {
+  const outcome: OutcomeEnumType = getOutcomeFromAuthcodeAndIsDirectAcquirer(authorizationCode, isDirectAcquirer);
+  mixpanel.track(PAYMENT_OUTCOME_CODE.value, {
```
Comment (pasqualespica) [inline src/response.ts:44]: Reminder : after this change we'd update [mixpanel funnel](https://eu.mixpanel.com/project/2319753/view/2865553/app/dashboards#id=1023737) too

**Review by pasqualespica — APPROVED** [review]
Great work 🚀

File: src/response.pug (hunk 1 above)
Comment (infantesimone) [reply src/response.pug:25]: good point, some `outcomes` are not mapped in the design and I think they cannot occur (they are errors between `PM` and `POS`). However it is a point to investigate and dedicate a separate PR.

**Pull request merged**